src/peerbook.ts

```diff
@@ -224,21 +217,22 @@ export class PeerbookConnection {
                 resolve(this.uid)
                 return
             }
-            if (!this.session) {
-                console.log("get UID with No session")
-                reject("No session")
-                return
-            }
-            this.session.sendCTRLMsg({type: "ping"}, (uid: string) => {
+            this.adminCommand({type: "ping"}).then((uid: string) => {
                 this.uid = uid
                 resolve(uid)
-            }, reject)
+            }).catch(reject)
         })
     }
 
     async connect(params?: ConnectParams) {
         return new Promise<void>((resolve, reject) =>{
```

Comment:
Consider passing an `async` function to the Promise

Reply:
We're stuck with Prmise as SendCtrlMessage is not a proper async. I've added #439